src/components/ActivityFeed/ActivityFeed.js

```diff
@@ -138,38 +138,64 @@ type State = {
   selectedEventData: ?Object | ?Transaction,
   eventType: string,
   eventStatus: string,
-}
-
-function getSortedFeedData(tabs, activeTab, feedData) {
-  if (tabs.length) {
-    const aTab = tabs.find(({ id, data = [] }) => id === activeTab && data.length) || { data: [] };
-    return aTab.data.sort((a, b) => b.createdAt - a.createdAt);
-  } else if (feedData.length) {
-    return feedData.sort((a, b) => b.createdAt - a.createdAt);
-  }
-  return [];
+  tabIsChanging: boolean,
+  formattedFeedData: Object[],
 }
 
 const PPNIcon = require('assets/icons/icon_PPN.png');
 
 class ActivityFeed extends React.Component<Props, State> {
-  feedData: Object[];
-
   static defaultProps = {
     initialNumToRender: 7,
   };
 
   constructor(props: Props) {
     super(props);
-    this.feedData = [];
     this.state = {
       showModal: false,
       selectedEventData: null,
       eventType: '',
       eventStatus: '',
+      tabIsChanging: false,
+      formattedFeedData: [],
     };
   }
 
+  componentDidMount() {
+    this.generateFeedSections();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const { tabs = [], feedData = [] } = this.props;
+    if ((tabs.length && !isEqual(tabs, prevProps.tabs))
```

Comment:
What if comparing without length check? Should be fine and would look nicer. Or are there any other concerns?

Reply:
Property `tabs` is optional. Hence, if it's not provided, "if" check will return false immediately, without a need to call `isEqual`